components/calendar/src/calendar_arithmetic.rs

```diff
@@ -102,11 +96,43 @@ impl ToExtendedYear for i32 {
     }
 }
 
+pub trait Pack: Copy {
+    type Packed: Copy + Debug;
+
+    fn pack(self, month: u8, day: u8) -> Self::Packed;
+    fn unpack_year(packed: Self::Packed) -> Self;
+    fn unpack_month(packed: Self::Packed) -> u8;
+    fn unpack_day(packed: Self::Packed) -> u8;
+}
+
+impl Pack for i32 {
+    type Packed = [u8; 4];
+
+    fn pack(self, month: u8, day: u8) -> Self::Packed {
+        (self << 9 | (month as i32) << 5 | day as i32).to_le_bytes()
+    }
+
+    fn unpack_year(packed: Self::Packed) -> Self {
+        let packed = i32::from_le_bytes(packed);
+        packed >> 9
+    }
+
+    fn unpack_month(packed: Self::Packed) -> u8 {
+        let packed = i32::from_le_bytes(packed);
+        (packed >> 5 & 0b1111) as u8
+    }
+
+    fn unpack_day(packed: Self::Packed) -> u8 {
+        let packed = i32::from_le_bytes(packed);
+        (packed & 0b11111) as u8
+    }
+}
+
 /// Trait for converting from era codes, month codes, and other fields to year/month/day ordinals.
 pub(crate) trait DateFieldsResolver: Calendar {
     /// This stores the year as either an i32, or a type containing more
     /// useful computational information.
-    type YearInfo: Copy + Debug + PartialEq + ToExtendedYear;
+    type YearInfo: Copy + Debug + PartialEq + ToExtendedYear + Pack;
```

Comment:
Suggestion: It seems that this is no longer `YearInfo`; now it is `YearMonthDayInfo`

Reply:
no it's not, it's still the year as either an `i32` or `HebrewYear`/`HijriYear` etc.. it allows being combined with a month and day to return a packed value, but this type is a year type

Reply:
ok. Maybe change the trait `Pack` to be called `PackAsYMD` or `PackWithMonthDay` or similar. I think I was confused because it looks like `YearInfo` is the thing that gets packed based on the trait name, but the packed thing is not the YearInfo, it is the YearInfo plus the month and day.